sdk/servicebus/Azure.Messaging.ServiceBus/tests/Processor/SessionProcessorLiveTests.cs

```diff
@@ -1939,6 +1939,69 @@ async Task ProcessMessage(ProcessSessionMessageEventArgs args)
             }
         }
 
+        [Test]
+        public async Task DiagnosticsPropertyThrowsWhenDiagnosticsNotEnabled()
+        {
+            var lockDuration = ShortLockDuration;
+            await using (var scope = await ServiceBusScope.CreateWithQueue(enablePartitioning: false, enableSession: true, lockDuration: lockDuration))
+            {
+                await using var client = CreateClient();
+                await using var processor = client.CreateSessionProcessor(scope.QueueName);
+
+                var tcs = new TaskCompletionSource<bool>();
```

Comment:
My inner Pavel is yelling at me here...

```suggestion
                var tcs = new TaskCompletionSource<bool>(TaskCreationOptions.RunContinuationsAsynchronously);
```

Reply:
Leaving one comment, but it seems to apply to many of the new test cases.